packages/office-ui-fabric-react/src/utilities/selection/examples/Selection.Basic.Example.tsx

```diff
@@ -81,160 +74,130 @@ const SelectionItemExample: React.FunctionComponent<ISelectionItemExampleProps>
   );
 };
 
-/**
- * The SelectionBasicExample controls the selection state of all items
- */
-export class SelectionBasicExample extends React.Component<{}, ISelectionBasicExampleState> {
-  private _hasMounted: boolean;
-
-  constructor(props: {}) {
-    super(props);
-
-    this._hasMounted = false;
-    // Memoizing this means that given the same parameters, it will return the same array of command objects
-    // (performance optimization)
-    this._getCommandItems = memoizeFunction(this._getCommandItems);
-
-    this.state = {
-      items: createListItems(ITEM_COUNT),
-      selection: new Selection({ onSelectionChanged: this._onSelectionChanged }),
-      canSelect: 'all',
-    };
-    this.state.selection.setItems(this.state.items, false);
-  }
-
-  public componentDidMount(): void {
-    this._hasMounted = true;
-  }
-
-  public render(): JSX.Element {
-    const { items, selection, canSelect } = this.state;
-
-    return (
-      <div className="ms-SelectionBasicExample">
-        <CommandBar items={this._getCommandItems(selection.mode, canSelect)} />
-        <MarqueeSelection selection={selection} isEnabled={selection.mode === SelectionMode.multiple}>
-          <SelectionZone selection={selection} onItemInvoked={this._alertItem}>
-            {items.map((item: IExampleItem, index: number) => (
-              <SelectionItemExample key={item.key} item={item} itemIndex={index} selection={selection} />
-            ))}
-          </SelectionZone>
-        </MarqueeSelection>
-      </div>
-    );
-  }
+export const SelectionBasicExample: React.FunctionComponent = () => {
+  const [canSelect, setCanSelect] = React.useState<'all' | 'vowels'>('all');
+  const forceUpdate = useForceUpdate();
+  const items = useConst<IExampleItem[]>(() => createListItems(ITEM_COUNT));
+  const [selection, setSelection] = React.useState<Selection>(() => {
```

Comment:
This can be simplified: 
```
const [selection, setSelection] = React.useState<Selection>(() => new Selection({
  onSelectionChanged: forceUpdate,
  items
});
```